src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.NativeAot.cs

```diff
@@ -716,18 +716,12 @@ public unsafe IntPtr GetOrCreateComInterfaceForObject(object instance, CreateCom
         {
             ArgumentNullException.ThrowIfNull(instance);
 
-            ManagedObjectWrapperHolder? managedObjectWrapper;
-            if (_managedObjectWrapperTable.TryGetValue(instance, out managedObjectWrapper))
+            ManagedObjectWrapperHolder managedObjectWrapper = _managedObjectWrapperTable.GetOrAdd(instance, static (c, items) =>
             {
-                managedObjectWrapper.AddRef();
-                return managedObjectWrapper.ComIp;
-            }
-
-            managedObjectWrapper = _managedObjectWrapperTable.GetValue(instance, (c) =>
-            {
-                ManagedObjectWrapper* value = CreateManagedObjectWrapper(c, flags);
+                ManagedObjectWrapper* value = items.This.CreateManagedObjectWrapper(c, items.Flags);
                 return new ManagedObjectWrapperHolder(value, c);
-            });
+            }, (This: this, Flags: flags));
```

Comment:
Use of ValueTuple in corelib has been scrutinized in the past for its impact on size. But given this is already in the weeds with ComWrappers, maybe we're more ok with it here?

Reply:
I couldn't think of a way to pass both parameters we need to the lambda without causing a new generic instantiation for this method (as in, either with this value tuple, or with some custom type, or whatever). I figured it would be an acceptable tradeoff here since the extra instantiation is only for this one method, and in return we can make this whole thing faster and zero-alloc? cc. @jkoritzinsky

Reply:
> without causing a new generic instantiation for this method (as in, either with this value tuple, or with some custom type, or whatever)

If we're fine with it here, that's fine.

But historically it's been more about when you get a generic instantation, what comes along with it, e.g. `KeyValuePair<>` has much less stuff on it that can't be trimmed away than does `ValueTuple<T1, T2>`.

Reply:
What if I used a local type instead, so we can avoid bringing along any other value type stuff (eg. `ITuple`)?

Reply:
If Jeremy is fine with a tuple here, I'm fine with it.

Reply:
Synced with Jeremy offline, and switched this to a local type 🙂